tagstudio/src/qt/widgets/line_edit_widgets/base_line_edit.py

```diff
@@ -0,0 +1,254 @@
+from PySide6.QtCore import QEasingCurve, QEvent, QPropertyAnimation, Qt, QTimer
+from PySide6.QtGui import (
+    QColor,
+    QEnterEvent,
+    QFocusEvent,
+    QPainter,
+    QPaintEvent,
+    QPalette,
+)
+from PySide6.QtWidgets import (
+    QApplication,
+    QLineEdit,
+    QProxyStyle,
+    QStyle,
+    QStyleOption,
+    QWidget,
+)
+
+from ... import theme
+
+# NOTE: ProxyStyle is used to customize the drawing of the line edit widget's background. Overriding
+# the repaint method would be overly complex as it would require reimplementing the calculation
+# of the cursor, text, alignment, margins, selected text highlight, index of the text when
+# clicked on, and click and drag to select, etc. By using ProxyStyle, we only need to customize
+# the background drawing. The widget will take care of the other aspects.
```

Comment:
It's explaining why ProxyStyle is used, not what ProxyStyle does. that's why i decided to write it as a comment instead of docstring of ProxyStyle. should i add this to the BaseLineEdit class docstring?